src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -92,6 +93,7 @@
     ]
 )
 
+snowflake_dialect.sets("reserved_keywords").discard("USER")
```

Comment:
I'm a bit worried about the potential for breakage here given what you raised in #1345 

Until we can pull out Postgres from Snowflake (assuming we even want to do this - not familair enough with them to know), should we instead reset Snowflake to what it was before by deleting all the postgres keywords it will inherit (including the extra ones you've now added in this PR), and have it reimport them from ANSI - plus adding any of the keywords it used to get on top of that from the old postgres code?

A bit of work I admit, and not really fair to lump you, a postgres developer with presumably no interest in Snowflake, with this, but do worry about this being a breaking change if we don't do that.

Alternatively we could just set up the Snowflake keyword list as part of this PR if we can get a definitive list for them?

Reply:
I see what you mean, but think it may be out of scope of this PR. From my perspective, Snowflake is a certain amount of broken, and that doesn't change with this PR. Every time we have done Postgres PRs, this issue has been present. and Snowflake's dialect definition has mutated. Snowflake currently passes all of the defined tests, so there's nothing other than USER that is an obvious symptom.

Reply:
I've had a look at Snowflake before you and another contributor started all the great Postgres work recently and it was fairly basic. As such I think it might be best to pull out Snowflake from Postgres anyway before next release in case your other changes break that anyway. Working on a PR now, which we might merge first before this, to resolve this issue.

Reply:
Raised this: https://github.com/sqlfluff/sqlfluff/pull/1353

Got a little obsessed with closing out the issues while I was in there!